conductor/agent/fetch.py

```diff
@@ -0,0 +1,142 @@
+
+import logging
+import json
+import os
+import sh
+
+import requests
+
+
+log = logging.getLogger("agent.fetcher")
+log.setLevel(logging.DEBUG)
+#logging.basicConfig(level=logging.DEBUG) 
+
+
+IMAGES_DIR="/dvl/co/porto/plugins-build/plugins/conductor/agent/images"
+
+def create_volume(registry, image, tag="latest"):
+
+    layers = fetch_layers(registry, image, tag)
+
+    for _id in reversed(layers):
+        import_layer_whands(_id)
+
+    volume_path = "/place/cocaine_volumes/%s:%s"%(image,tag)
+    sh.mkdir("-p", volume_path)
+    print sh.portoctl("vcreate", volume_path, "layers="+(";".join(layers)))
```

Comment:
No, it's not, as far as I can tell. It can be done via RPC calls, though, so later we can use Porto proto files and async RPC.